File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/camera/Camera.java
package com.yanzhenjie.album.api.camera;

/**
 * 相机功能统一接口（契约）
 * 定义：相机必须提供【拍照】和【录像】两个能力
 *
 * @param Image   拍照包装类
 * @param Video   录像包装类
 */
public interface Camera<Image, Video> {

    /**
     * 拍照
     */
    Image image();

    /**
     * 录像
     */
    Video video();

}

File: lib_local/gallery/src/main/java/com/yanzhenjie/album/api/camera/Camera.kt
package com.yanzhenjie.album.api.camera

/**
 * Created by YanZhenjie on 2017/8/18.
 */
interface Camera<Image, Video> {

    /**
     * Take picture.
     */
    fun image(): Image

    /**
     * Take video.
     */
    fun video(): Video

}